### ISSUE TITLE
Connect to Kafka cluster only if new data comes in

### ISSUE THREAD

**Issue by CPS86** [Shopify/sarama#887]
Sarama Version: latest
Kafka Version: latest
Go Version: 1.8

Hello

I'm using the asyncproducer to send data and everything works as expected, however, I don't want to be connected to my Kafka cluster all the time.  Is it possible to connect to kafka cluster only if new data comes in? I was looking for some bool IsConnected func in sarama, unfortunately, without success. 
In other words, I would like to close the producer if it doesn't have any work to do and then open it again if new data will be available to digest.

How can i achieve such behaviour?

**Comment by CPS86** [comment]
Okay, and what about KeepAlive ?

   // KeepAlive specifies the keep-alive period for an active network connection.
        // If zero, keep-alives are disabled. (default is 0: disabled).
        KeepAlive time.Duration

So If I set keepAlive to = 10, it means that it will keep connections open ONLY for 10 seconds, unless new data will be available for sending? Can I use that in this matter?

**Comment by eapache** [comment]
No, the KeepAlive setting refers to the frequency of TCP KeepAlive (http://tldp.org/HOWTO/TCP-Keepalive-HOWTO/overview.html) heartbeat packets. If you set keepAlive to 10, it will send an extra packet every 10 seconds to ensure that the connection stays alive.

**Comment by CPS86** [comment]
Thanks, I understand now :)
